cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpProvisionSetup.java

```diff
@@ -74,14 +78,38 @@ public void execute(AuthenticatedContext authenticatedContext, CloudStack stack)
                 image.setRawDisk(rawDisk);
                 Compute.Images.Insert ins1 = compute.images().insert(projectId, image);
                 ins1.execute();
-                PollTask<Boolean> task = new GcpImageCheckerTask(authenticatedContext, projectId, image.getName(), compute);
-                syncPollingScheduler.schedule(task);
             }
         } catch (Exception e) {
             String msg = String.format("Error occurred on %s stack during the setup: %s", stackId, e.getMessage());
             LOGGER.error(msg, e);
             throw new CloudConnectorException(msg, e);
         }
+    }
+
+    @Override
+    public CheckImageProgress checkImageStatus(AuthenticatedContext authenticatedContext, CloudStack stack) {
+        CloudCredential credential = authenticatedContext.getCloudCredential();
+        String projectId = getProjectId(credential);
+        String imageName = stack.getImage().getImageName();
+        try {
+            Image image = new Image();
+            image.setName(getImageName(imageName));
+            Compute compute = buildCompute(credential);
+            Compute.Images.Get getImages = compute.images().get(projectId, image.getName());
+            String status = getImages.execute().getStatus();
+            LOGGER.info("Status of image {} copy: {}", image.getName(), status);
+            if (READY.equals(status)) {
+                return new CheckImageProgress(ImageStatus.CREATE_FINISHED, ONE_HUNDRED);
+            }
+        } catch (IOException e) {
+            LOGGER.warn("Failed to retrieve image copy status", e);
+            return new CheckImageProgress(ImageStatus.CREATE_FAILED, 0);
+        }
+        return new CheckImageProgress(ImageStatus.IN_PROGRESS, HALF);
```

Comment:
Wouldn't it be possible to calculate the exact percentage based on the copied bytes?